README.md

```diff
@@ -1,44 +1,225 @@
+    );
+
+    // Print GET response data
+    print('GET Response: ${response.data}');
+  } catch (e) {
+    // Handle GET request error
+    print('GET Error: $e');
+  }
+}
+
+// Function to make a sample PUT request
+void makePutRequest(Dio dio) async {
+  try {
+    // Define PUT request data
+    final requestData = {'updated_key': 'updated_value'};
+
+    // Make PUT request
+    final response = await dio.put(
+      'https://api.example.com/endpoint', // Replace with your actual PUT endpoint URL
+      data: requestData, // Request data to be sent
+    );
+
+    // Print PUT response data
+    print('PUT Response: ${response.data}');
+  } catch (e) {
+    // Handle PUT request error
+    print('PUT Error: $e');
+  }
+}
+
+// Function to make a sample DELETE request
+void makeDeleteRequest(Dio dio) async {
+  try {
+    // Make DELETE request
+    final response = await dio.delete(
+      'https://api.example.com/endpoint', // Replace with your actual DELETE endpoint URL
+    );
+
+    // Print DELETE response data
+    print('DELETE Response: ${response.data}');
+  } catch (e) {
+    // Handle DELETE request error
+    print('DELETE Error: $e');
+  }
+}
+```
 
-{Project tag line}
+## Features
 
-{Small description of the purpose of the project}
+### Default Header signature and algorithm
 
-[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](LICENSE)
+By default, here is how the header is constructed:
 
-## Getting Started
+The final header has the following specification: `{ClientId}:{Nonce}:{Timestamp}:{SignatureBody}` where:
+- `{ClientId}`: is the client id as specified by the configuration
+- `{Nonce}`: is a random value unique to each request (a UUID/GUID is perfectly suitable)
+- `{Timestamp}`: is the current time when the request is sent, in Unix Epoch time (in seconds)
+- `{SignatureBody}`: Is the Base-64 encoded value of the HMAC SHA256 Signature of the signature components
 
-{Instructions to quickly get started using the project: pre-requisites, packages
-to install, sample code, etc.}
+Signature components (the source for the SignatureBody HMAC value) is a binary value composed of the following values sequentially:
+- Nonce: UTF-8 encoded binary values of the Nonce
+- Timestamp: UTF-8 encoded binary values of the Timestamp (as a string value)
+- Request method: UTF-8 encoded binary values of the **uppercase** Request method
+- Request scheme: UTF-8 encoded binary values of the Request Uri scheme (e.g. `https`)
+- Request host: UTF-8 encoded binary values of the Request Uri host (e.g. `example.org`)
+- Request local path: UTF-8 encoded binary values of the Request Uri local path (e.g. `/api/v1/users`)
+- Request query string: UTF-8 encoded binary values of the Request Query string, including the leading `?` (e.g. `?q=search`)
+- Request body: Raw bytes of the request body
 
-## Features
+*See the Configuration section on how to customize the signature.*
+
+### Nonce repository
```

Comment:
Is this section relevant for the Dart version? I seems to be specific to the AspNetCore part.